architecture/2024-08-applications-rp-component-threat-model.md

```diff
@@ -0,0 +1,224 @@
+
+2. Applications RP also uses UCP's queue and store libraries. These libraries use Cryptography and are detailed in UCP threat model. [link to be added]
+
+#### Storage of secrets
+
+Applications RP has access to sensitive information related to the application resource it manages as well as the cloud credentials it requires for managing cloud resources on Azure and AWS. Applications RP provides a Secret Store which can be used to store sensitive information such as TLS certificate and private keys. It uses kubernetes secrets to implement this secret store. 
+
+##### Managing secrets for datastores
+
+Applications RP service has a Datastore RP. This RP is the resource provider for datstores such as SQL database, Mongo DB and Redis Cache. As of today, sensitive information such as DB connection string, user/ password that is required to provision these resources is stored in plain text. The feature which enables datastores to use a secret store is in progress. 
+
+##### Access to cloud credentials and cloud
+
+Applications RP requires AWS and Azure credentials for accessing and managing resources in cloud. It fetches credentials using UCP Secret Provider library. 
+Credentials are not available for retrieval through API. Also, RP supports federated identities which enables robust security practices. 
+
+Using these credentials, Applications RP can create other resources in AWS and Azure. The RP also can create managed identities for azure which will decide who can deploy and run user code.  
+
+#### Access to cluster
+
+The RP can create kubernetes resources and manage them on behalf of the user. It can, for example, create a container based on the image provided by the user, which can in turn execute arbitrary code, and create other resources in the cluster where Radius is running, or on a different cluster. 
+
+#### Exposing User Application to Internet
+
+The Radius RP can create ingress Kubernetes objects, which can expose a kubernetes service to internet. 
+
+#### Bicep Recipe execution
+
+`Recipes` are arbitrary Bicep or Terraform code which can provision infrastructure resources. In order to exceute a Bicep recipe, Applications RP's Recipe Engine first downloads the recipe from a OCI compliant registry over HTTP. Then it requests UCP to deploy the bicep. 
+
+#### Terraform Recipe execution
+
+Terraform recipes are download from internet too, since they are available as public modules. These recipes are downloaded onto an empty directory `/terraform` which is mounted into the applications RP pod. Then the installed terraform executable executes these recipes in the current directory. Terraform communicates with AWS and Azure as needed to deploy resources.
```

Comment:
We support private modules as well as different kinds of authentication.

Terraform downloads providers from the internet and executes them inside Applications RP's pod. This means that they have access to the network, filesystem, and environment variables, as well as the running memory of applications rp.